rclpy/rclpy/executors.py

```diff
@@ -1066,6 +1066,9 @@ def _spin_once_impl(
                         self._futures.remove(future)
                         future.result()  # raise any exceptions
 
+            # Yield GIL so executor threads have a chance to run.
+            time.sleep(0)
```

Comment:
How about this? can be more efficient. it directly tells the OS scheduler to yield the CPU to another thread without any sleep/timer overhead, and it clearly communicates the intent (yield CPU) rather than sleeping.

```suggestion
            os.sched_yield() if hasattr(os, 'sched_yield') else time.sleep(0)
```

Reply:
Done